lib/streamlit/caching/memo_decorator.py

```diff
@@ -392,36 +401,41 @@ def get_stats(self) -> List[CacheStat]:
                 )
         return stats
 
-    def read_value(self, key: str) -> Any:
-        """Read a value from the cache. Raise `CacheKeyNotFoundError` if the
-        value doesn't exist, and `CacheError` if the value exists but can't
+    def read_value(self, key: str) -> CachedResult:
+        """Read a value and messages from the cache. Raise `CacheKeyNotFoundError`
+        if the value doesn't exist, and `CacheError` if the value exists but can't
         be unpickled.
         """
         try:
-            pickled_value = self._read_from_mem_cache(key)
+            pickled_entry = self._read_from_mem_cache(key)
 
         except CacheKeyNotFoundError as e:
             if self.persist == "disk":
-                pickled_value = self._read_from_disk_cache(key)
-                self._write_to_mem_cache(key, pickled_value)
+                pickled_entry = self._read_from_disk_cache(key)
+                self._write_to_mem_cache(key, pickled_entry)
             else:
                 raise e
 
         try:
-            return pickle.loads(pickled_value)
+            entry = pickle.loads(pickled_entry)
+            assert isinstance(entry, CachedResult)
```

Comment:
This should just be a nonfatal check that throws a CacheError as if there was an UnpicklingError